google/cloud/storage/blob.py

```diff
@@ -1667,7 +1669,7 @@ def _do_resumable_upload(
         )
 
         while not upload.finished:
-            response = upload.transmit_next_chunk(transport, timeout=timeout)
+            response = telemetry_wrapped_api_request(upload.transmit_next_chunk, transport, timeout=timeout, instrumented_function_name='upload')
```

Comment:
could you explain why in some cases we call the wrapped function and provide args, but in other cases we are using a partial? It seems a bit strange to me that both varieties are being used, but I think I may be missing a subtle detail.

Reply:
Most client functions will eventually call `client._connection.api_request`. The exceptions to this are `_do_download` and `upload` which will, instead of using `client._connection.api_request`, pass in the raw transport and use another unrelated method. I'm a bit fuzzy on the details maybe @frankyn can explain better.